query/src/main/java/jetbrains/exodus/query/InstanceOf.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright ${inceptionYear} - ${year} ${owner}
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.exodus.query;
+
+import jetbrains.exodus.entitystore.Entity;
+import jetbrains.exodus.entitystore.orientdb.iterate.property.OInstanceOfIterable;
+import jetbrains.exodus.query.metadata.ModelMetaData;
+
+public class InstanceOf extends NodeBase {
```

Comment:
Would be great to have a dedicated test for the node within OQueryEngineTest

Reply:
There is one.
fun `instance of should work`()